code/modules/antagonists/heretic/heretic_knowledge.dm

```diff
@@ -184,6 +184,8 @@
 	user.mind.AddSpell(spell_to_add)
 
 /datum/heretic_knowledge/spell/on_lose(mob/user)
+	if(!user.mind)
+		return
```

Comment:
this is another one of the runtimes i got, not quite sure why this happens, might be because during testing i stacked a lot of antags on top of each other, will try with only heretic though

Reply:
@MrMelbert there is an issue with heretics transferring bodies, its not an AI-exclusive thing, kept getting these 2 runtimes where it asks for the mind of the old body but that gets set to null in /datum/mind/transfer_to()

Reply:
removed all the unnecessary changes to files @MrMelbert